app/js/ui.js

```diff
@@ -38,10 +44,60 @@ class Controller {
   }
 
   initTemplateBlocks() {
-    const blocks = document.querySelectorAll('.block.template');
-    for (let i = 0; i < blocks.length; i++) {
-      const block = blocks[i];
-      this.templateBlocks.push(new TemplateBlock(block));
+    const templateBlocks = document.querySelectorAll('.block.template');
+    for (let i = 0; i < templateBlocks.length; i++) {
+      const block = templateBlocks[i];
+      this.templateBlocks.push(new blocks.TemplateBlock(block));
+    }
+  }
+
+  createBlock(blockType, x, y) {
+    let block;
+    switch (blockType) {
+      case 'Loop':
+        block = new blocks.LoopBlock(x, y);
+        break;
+      case 'Conditional':
+        block = new blocks.ConditionalBlock(x, y);
+        break;
+      case 'Variable':
+        block = new blocks.VariableBlock(x, y);
+        break;
+      default:
+        throw new TypeError('Unrecognized block type');
+    }
+    this.canvas.addBlock(block);
+    const selectBlock = () => {
+      if (this.activeBlock) {
+        this.activeBlock.toggleSelect();
+      }
+      block.toggleSelect();
+      this.activeBlock = block;
+      document.getElementById('config-bar')
+        .getElementsByTagName('content')[0]
+        .innerHTML = this.activeBlock.getParamsMeta();
+    };
+    block.element.addEventListener('click', selectBlock, false);
+    this.configBarActive = false;
+    this.toggleConfigBar();
+    selectBlock();
+  }
+
+  toggleConfigBar() {
+    const configBar = document.getElementById('config-bar');
+    const configContent = document.getElementById('config-bar').getElementsByTagName('content')[0];
+    const mainWindow = document.getElementsByTagName('main')[0];
+    const viewportWidth = document.documentElement.clientWidth;
+    if (this.configBarActive) {
```

Comment:
My preferred way to handle style updates like this is to assign a class to the parent. I.e. give the body a "sidebar" or "blockbar" class and keeping style changes in the css.